### PR TITLE
Add helper class to deserialize server requests

### PR DESCRIPTION
**Pull request by Arun-LinkedIn**
Adding helper class to deserialize server requests. This will be helpful to deserialize incoming request and send response from Netty IO threads itself in future PR.

### CODE REVIEW

File: ambry-protocol/src/main/java/com/github/ambry/protocol/ServerRequestResponseHelper.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        default:
+          throw new UnsupportedOperationException("Request type not supported");
+      }
+    } else {
+      request = (RequestOrResponse) ((LocalRequestResponseChannel.LocalChannelRequest) networkRequest).getRequestInfo()
+          .getRequest();
+      if (request.getRequestType() == RequestOrResponseType.PutRequest) {
+        // For PutRequest sent via local channel, we have to create a new PutRequest object to wrap the data buffer as a
+        // stream. Also, we need to set the crc value to null since the request is not coming over network.
+        PutRequest receivedPutRequest = ((PutRequest) request);
```
Comment (JingQianCloud) [inline ambry-protocol/src/main/java/com/github/ambry/protocol/ServerRequestResponseHelper.java:83]: For the PutBlob local request, getDecodedRequest will return the newly created PutRequest. 
When we send out the response for that request, we'll use this internal created PutRequest as the reference.
But it's not a LocalChannelRequest, right? Do we have problem for the sendResponse as below?
In the old implementation, although we also create the new PutBlob, but we send out the response based on the original request.

```
    @Override
  public void sendResponse(Send payloadToSend, NetworkRequest originalRequest, ServerNetworkResponseMetrics metrics) {
    LocalChannelRequest localRequest = (LocalChannelRequest) originalRequest;
    ResponseInfo responseInfo =
        new ResponseInfo(localRequest.requestInfo, null, localRequest.requestInfo.getReplicaId().getDataNodeId(),
            payloadToSend);
    BatchBlockingQueue<ResponseInfo> responseQueue = getResponseQueue(localRequest.processorId);
    responseQueue.put(responseInfo);
    logger.debug("Added response for {}, size now {}", localRequest.processorId, responseQueue.size());
  }

```

Comment (Arun-LinkedIn) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/ServerRequestResponseHelper.java:83]: Hmm, actually PutRequest is present inside `NetworkRequest` as `networkRequest.requestInfo.request`. In both old and new code, we are using original `NetworkRequest` in the `sendResponse()`. I think creating a new `PutRequest` should be okay.

Comment (JingQianCloud) [reply ambry-protocol/src/main/java/com/github/ambry/protocol/ServerRequestResponseHelper.java:83]: I see. With the new code, we still send out the response with the NetworkRequest. I missed that.

**Review by JingQianCloud — APPROVED** [review]

**Comment by Arun-LinkedIn** [comment]
Thank you @JingQianCloud . Merging the PR so that I can put up dependent PR.

**Pull request merged**